central/detection/lifecycle/manager_impl.go

```diff
@@ -286,22 +315,63 @@ func (m *managerImpl) checkAndUpdateBaseline(baselineKey processBaselineKey, ind
 		insertableElement := &storage.BaselineItem{Item: &storage.BaselineItem_ProcessName{ProcessName: baselineItem}}
 		elements = append(elements, insertableElement)
 	}
-	if len(elements) == 0 {
+
+	if checkIfBaselineCanBeSkipped(elements, inObservation, baseline) {
 		return false, nil
 	}
+
 	if !exists {
-		_, err = m.baselines.UpsertProcessBaseline(lifecycleMgrCtx, key, elements, true, true)
+		userLock := autolockEnabled && !inObservation
+		upsertedBaseline, err := m.baselines.UpsertProcessBaseline(lifecycleMgrCtx, key, elements, true, true, userLock)
+		if err != nil {
+			return false, err
+		}
+		if userLock {
+			err = m.SendBaselineToSensor(upsertedBaseline)
+		}
 		return false, err
 	}
 
 	userBaseline := processbaseline.IsUserLocked(baseline)
 	roxBaseline := processbaseline.IsRoxLocked(baseline) && hasNonStartupProcess
-	if userBaseline || roxBaseline {
+	reprocessRisk := userBaseline || roxBaseline
+
+	if reprocessRisk {
 		// We already checked if it's in the baseline and it is not, so reprocess risk to mark the results are suspicious if necessary
 		m.reprocessor.ReprocessRiskForDeployments(baselineKey.deploymentID)
-	} else {
-		// So we have a baseline, but not locked.  Now we need to add these elements to the unlocked baseline
-		_, err = m.baselines.UpdateProcessBaselineElements(lifecycleMgrCtx, key, elements, nil, true)
+	}
+
+	if !autolockEnabled {
+		if !reprocessRisk {
+			_, err := m.baselines.UpdateProcessBaselineElements(lifecycleMgrCtx, key, elements, nil, true, false)
+			if err != nil {
+				return false, err
+			}
+		}
+		return userBaseline, nil
+	}
+
+	// If this point is reached AutoLockProcessBaselines is enabled.
```

Comment:
Doesn't this section violate our discussion of allowing the user to unlock an auto locked baseline?  If I'm following this code correctly, if a baseline exists as unlocked but autolock is on, this will lock it.  I don't think that is what we want.  As we stated in the meeting on August 26 we only want to lock baselines from point forward.